src/constraint/field.ts

```diff
@@ -243,7 +243,7 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
               // ignore type and properties of wildcards
               continue;
             }
-            let isSupported = channelScalePropertyIncompatability(channel, scaleProp as keyof Scale) === undefined;
+            let isSupported = channelScalePropertyIncompatability(channel as Channel, scaleProp as keyof Scale) === undefined;
```

Comment:
Why do we need to force channel to be a Channel? Can we support ExtendedChannel in channelScalePropertyIncompatability?